Quotient/events/roommessageevent.cpp

```diff
@@ -86,49 +88,47 @@ inline bool isReplacement(const std::optional<EventRelation>& rel)
 
 QJsonObject RoomMessageEvent::assembleContentJson(const QString& plainBody,
                                                   const QString& jsonMsgType,
-                                                  TypedBase* content)
+                                                  TypedBase* content,
+                                                  std::optional<EventRelation> relatesTo)
 {
     QJsonObject json;
     if (content) {
         // TODO: replace with content->fillJson(json) when it starts working
         json = content->toJson();
-        if (jsonMsgType != TextTypeKey && jsonMsgType != NoticeTypeKey
-            && jsonMsgType != EmoteTypeKey) {
-            if (json.contains(RelatesToKey)) {
-                json.remove(RelatesToKey);
-                qCWarning(EVENTS)
-                    << RelatesToKey << "cannot be used in" << jsonMsgType
-                    << "messages; the relation has been stripped off";
+    }
+    json.insert(MsgTypeKey, jsonMsgType);
+    json.insert(BodyKey, plainBody);
+    if (relatesTo.has_value()) {
+        json.insert(RelatesToKey, toJson<EventRelation>(relatesTo.value()));
+        // JsonObjectConverter<EventRelation>::dumpTo(json, relatesTo.value());
+        if (auto* textContent = static_cast<const TextContent*>(content); relatesTo->type == EventRelation::ReplacementType) {
+            QJsonObject newContentJson;
+            if (content && textContent->mimeType.inherits("text/html"_L1)) {
+                newContentJson.insert("format"_L1, HtmlContentTypeId);
```

Comment:
```suggestion
                newContentJson.insert(FormatKey, HtmlContentTypeId);
```
Assuming the comment about `FormatKey` definition is applied.